ui/src/app/components/resource-actions-menu.tsx

```diff
@@ -16,9 +16,13 @@ export interface Action {
 export const ResourceActionsMenu = (props: {
   actions: Action[];
   disabled?: boolean;
-  appsAnalysis?: boolean;
+  menuButtonComponentOverride?: ({ ...props }) => JSX.Element;
 }) => {
-  const { actions, disabled } = props;
+  const { actions, disabled, menuButtonComponentOverride } = props;
+
+  const menuButtonComponent: ({ ...props }) => JSX.Element =
+    menuButtonComponentOverride ?? SnowmanButton;
```

Comment:
We don't use `??` much.  It's the "nullish coalesce operator" and means "use the first thing if it exists, else use the second thing"